addons/mass_mailing/models/mailing.py

```diff
@@ -1199,3 +1250,29 @@ def _image_to_url(b64image: bytes):
             return lxml.html.tostring(root)
 
         return body_html
+
+    @api.model
+    def get_favorites(self):
+        """Return all mailing set as favorite and skip mailing with empty body."""
+        values_list = self.search_read(
+            domain=[
+                ('body_arch', '!=', False),
+                ('body_html', '!=', False),
+                ('favorite', '=', True),
+            ],
+            fields=['id', 'subject', 'body_arch', 'user_id', 'mailing_model_id'],
+            order='favorite_date DESC',
+        )
+
+        def _sort_mailing(values):
+            """You see first the mailing without responsible, then your mailing and then the others."""
+            if not values['user_id']:
+                return -1
+            return values['user_id'][0] != self.env.user.id
+
+        values_list.sort(key=_sort_mailing)
+
+        return [
+            values for values in values_list
+            if not tools.is_html_empty(values['body_arch'])
```

Comment:
Arf you are doing it already :) would do it before sorting (search mailings, filter out void, then sort and return). Makes more sense that way.